File: web/src/plugin/cert_manager/api/certCategory.js
import service from '@/utils/request'

export const createCertCategory = (data) => {
  return service({
    url: '/certCategory/createCertCategory',
    method: 'post',
    data
  })
}

export const deleteCertCategory = (data) => {
  return service({
    url: '/certCategory/deleteCertCategory',
    method: 'delete',
    data
  })
}

export const updateCertCategory = (data) => {
  return service({
    url: '/certCategory/updateCertCategory',
    method: 'put',
    data
  })
}

export const getCertCategoryList = (params) => {
  return service({
    url: '/certCategory/getCertCategoryList',
    method: 'get',
    params
  })
}

export const batchUpdateDomainCategory = (data) => {
  return service({
    url: '/certCategory/batchUpdateDomainCategory',
    method: 'post',
    data
  })
}

File: server/plugin/cert_manager/service/cert_category.go
package service

import (
	"github.com/flipped-aurora/gin-vue-admin/server/global"
	"github.com/flipped-aurora/gin-vue-admin/server/plugin/cert_manager/model"
	"github.com/flipped-aurora/gin-vue-admin/server/plugin/cert_manager/model/request"
)

type certCategory struct{}

func (s *certCategory) CreateCertCategory(cc model.CertCategory) error {
	return global.GVA_DB.Create(&cc).Error
}

func (s *certCategory) DeleteCertCategory(cc model.CertCategory) error {
	return global.GVA_DB.Delete(&cc).Error
}

func (s *certCategory) UpdateCertCategory(cc model.CertCategory) error {
	return global.GVA_DB.Save(&cc).Error
}

func (s *certCategory) GetCertCategory(id uint) (cc model.CertCategory, err error) {
	err = global.GVA_DB.Where("id = ?", id).First(&cc).Error
	return
}

func (s *certCategory) GetCertCategoryList(info request.CertCategorySearch) (list interface{}, total int64, err error) {
	limit := info.PageSize
	offset := info.PageSize * (info.Page - 1)
	db := global.GVA_DB.Model(&model.CertCategory{})
	var categories []model.CertCategory
	if info.Name != "" {
		db = db.Where("name LIKE ?", "%"+info.Name+"%")
	}
	err = db.Count(&total).Error
	if err != nil {
		return
	}
	err = db.Limit(limit).Offset(offset).Find(&categories).Error
	return categories, total, err
}

func (s *certCategory) BatchUpdateCertCategory(info request.BatchUpdateCategory) error {
	return global.GVA_DB.Model(&model.CertCertificate{}).Where("id IN ?", info.IDs).Update("category", info.Category).Error
}
